server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -80,8 +80,22 @@ class InteractiveSession(
 
     def mergeConfList(list: Seq[String], key: String): Unit = {
       if (list.nonEmpty) {
-        val newList = (list ++ conf.get(key)).mkString(",")
-        builder.setConf(key, newList)
+        val oldList = builder.getConf(key)
```

Comment:
The reason I expose LivyClientBuilder.getConf is that mergeConfList will have issue when being called multiple times for the same key.  e.g. I want to call mergeConfList multiple times for spark.jars, For the first call, it would update builder conf correctly, but for the second call, it would lose the first update because it didn't get the old values from LivyClientBuilder.